### PR TITLE
add `to` method for PackedSequence

### LINKED ISSUE #7221: PackedSequence is missing `to` method
The following will die as PackedSequence doesn't have `to` method
```(python)
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
torch.nn.utils.rnn.pad_sequence(sequences).to(device)
```
So instead one has to do old-fashioned `.cuda()` call.

### PR DESCRIPTION
**Pull request by facaiy**
Fix #7221.

It's my first contribution to pytorch, so any help would be appreciated. Thanks.

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -161,6 +161,24 @@ def err_fn():
...
 
+    def test_to(self):
+        padded, lengths = self._padded_sequence(torch.IntTensor)
+        a = rnn_utils.pack_padded_sequence(padded, lengths).cpu()
+
+        self.assertIs(a, a.to('cpu'))
+        self.assertIs(a, a.to('cpu', dtype=torch.int32))
+        self.assertEqual(a.long(), a.to(torch.int64))
+
+        if torch.cuda.data.is_available():
```
Comment (SsnL) [inline test/test_nn.py]: extra `data.`

File: torch/nn/utils/rnn.py [hunk 2]
```diff
@@ -85,6 +85,14 @@ def byte(self):
         r"""Returns copy with `self.data` cast to byte type"""
         return type(self)(self.data.byte(), self.batch_sizes)
 
+    def to(self, *args, **kwargs):
+        r"""Performs dtype and/or device conversion on `self.data`"""
```
Comment (SsnL) [inline torch/nn/utils/rnn.py]: Could you at least mention that this is similar to `` :meth:`torch.Tensor.to` ``?

**Comment by facaiy** [comment]
@SsnL Thanks for your review. I think I have resolved both your comments, could you take a look again? 
By the way, it seems that we can use `python test/run_test` to select specific test case locally, does it have any document?

**Review by ssnl — APPROVED** [review]
LGTM. Thanks @facaiy !

**Review by soumith — APPROVED** [review]

**Pull request merged**